src/err/mod.rs

```diff
@@ -602,10 +602,10 @@ mod tests {
             .split(", ");
 
         assert_eq!(fields.next().unwrap(), "type: <class 'Exception'>");
-        #[cfg(not(Py_3_7))] // Python 3.6 and below formats the repr differently
-        assert_eq!(fields.next().unwrap(), ("value: Exception('banana',)"));
-        #[cfg(Py_3_7)]
-        assert_eq!(fields.next().unwrap(), "value: Exception('banana')");
+        assert!(fields
+            .next()
+            .unwrap()
+            .starts_with("value: Exception('banana'"));
```

Comment:
Can we please not make this change?

I left the `#[cfg(...)]` variants in, even though it is more LOC, because it is clearer what the exact behaviour is. `starts_with` is a much weaker check than `assert_eq`.

When we drop Python 3.6 support (presumably next year) it is easy to update this test then to remove the `#[cfg]`s.

Reply:
I did this not for reducing LOC, but for passing tests with `abi3-py36` with Python >= 3.7.
I'm happy to go in another way, e.g., using actual Python version instead of cfg here, though I'm not sure wa can.

Reply:
Ah, sorry, I see now. With abi3 flags the `#[cfg()]` applies only `Py_3_6`, even on Python 3.7. 

I guess this is actually a runtime dependency on the Python version - something like `if / else` instead of `#[cfg]`...

Reply:
I have an idea for checking this at runtime that I'll push slightly later today.

Reply:
See #1322. I fixed these tests to use a new `Python::version()` API.